source/shared/core_sqlsrv.h

```diff
@@ -2352,4 +2354,26 @@ struct str_conn_attr_func {
     }
 };
 
+struct str_conn_attr_auth_func {
+
+    static void func( connection_option const* option, zval* value, sqlsrv_conn* conn, std::string& conn_str TSRMLS_DC )
+    {
+        try {
+            SQLSRV_ASSERT( Z_TYPE_P( value ) == IS_STRING, "str_conn_attr_auth_func: Unexpected zval type." );
+            
+            std::string aads = Z_STRVAL_P( value );
+            SQLSRV_ASSERT( aads != "ActiveDirectoryIntegrated",
```

Comment:
the error messages should be registered in error array, you can find similar definitions in util and pdo_util (https://github.com/Microsoft/msphpsql/blob/dev/source/sqlsrv/util.cpp#L51)